src/evaluators/utility/PHAL_ComputeBasisFunctionsSide_Def.hpp

```diff
@@ -228,77 +228,22 @@ operator() (const ComputeBasisFunctionsSide_Collapsed_Tag& tag, const int& sideS
 
-  // Computing the metric
-  for (unsigned int qp=0; qp<numSideQPs; ++qp) {
-    for (unsigned int idim=0; idim<numSideDims; ++idim) {
-      // Diagonal
-      metric(cell,side,qp,idim,idim) = 0.;
-      for (unsigned int coor=0; coor<numCellDims; ++coor) {
-        metric(cell,side,qp,idim,idim) += tangents(cell,side,qp,coor,idim)*tangents(cell,side,qp,coor,idim); // g = J'*J
-      }
+  for (std::size_t node=0; node < numNodes; ++node)
+    for (std::size_t dim=0; dim < numCellDims; ++dim)
+      physPointsCell(iCell, node, dim) = coordVec(sideSet.cellsOnSide(currentSide,iCell),node,dim);
 
-      // Extra-diagonal
-      for (unsigned int jdim=idim+1; jdim<numSideDims; ++jdim) {
-        metric(cell,side,qp,idim,jdim) = 0.;
-        for (unsigned int coor=0; coor<numCellDims; ++coor) {
-          metric(cell,side,qp,idim,jdim) += tangents(cell,side,qp,coor,idim)*tangents(cell,side,qp,coor,jdim); // g = J'*J
-        }
-        metric(cell,side,qp,jdim,idim) =  metric(cell,side,qp,idim,jdim);
-      }
-    }
-  }
+}
 
-  // Computing the metric determinant, the weighted measure and the inverse of the metric
-  switch (numSideDims) {
-    case 1:
-      for (unsigned int qp=0; qp<numSideQPs; ++qp) {
-        metric_det(cell,side,qp) =  metric(cell,side,qp,0,0);
-        w_measure(cell,side,qp) = cub_weights(qp)*std::sqrt( metric(cell,side,qp,0,0));
-        inv_metric(cell,side,qp,0,0) = 1./ metric(cell,side,qp,0,0);
-      }
-      break;
-    case 2:
-      for (unsigned int qp=0; qp<numSideQPs; ++qp) {
-        metric_det(cell,side,qp) =  metric(cell,side,qp,0,0)* metric(cell,side,qp,1,1) -  metric(cell,side,qp,0,1)* metric(cell,side,qp,1,0);
-        w_measure(cell,side,qp) = cub_weights(qp)*std::sqrt(metric_det(cell,side,qp));
-        inv_metric(cell,side,qp,0,0) =  metric(cell,side,qp,1,1)/metric_det(cell,side,qp);
-        inv_metric(cell,side,qp,1,1) =  metric(cell,side,qp,0,0)/metric_det(cell,side,qp);
-        inv_metric(cell,side,qp,0,1) = inv_metric(cell,side,qp,1,0) = - metric(cell,side,qp,0,1)/metric_det(cell,side,qp); 
-      }
-      break;
-    default:
-      break;
-  }
+template<typename EvalT, typename Traits>
+KOKKOS_INLINE_FUNCTION
+void ComputeBasisFunctionsSide<EvalT, Traits>::
+operator() (const GatherNormals_Tag& tag, const int& iCell) const {
 
-  for (unsigned int node=0; node<numSideNodes; ++node) {
-    for (unsigned int qp=0; qp<numSideQPs; ++qp) {
-      for (unsigned int ider=0; ider<numSideDims; ++ider) {
-        GradBF(cell,side,node,qp,ider)=0;
-        for (unsigned int jder=0; jder< numSideDims; ++jder) {
-          GradBF(cell,side,node,qp,ider) += inv_metric(cell,side,qp,ider,jder)*grad_at_cub_points(node,qp,jder);
-        }
-      }
-    }
-  }
+  for (unsigned int icoor=0; icoor<numCellDims; ++icoor)
+    for (unsigned int qp=0; qp<numSideQPs; ++qp)
+      normals(sideSet.sideSetIdxOnSide(currentSide,iCell),qp, icoor) = normals_view(iCell,qp,icoor);
```

Comment:
this is a race condition if multiple cells touch a single "sideset index on side". I'm not sure if that's the case here.

Reply:
The sideset indices for each side are guaranteed to be disjoint sets so this one is safe.

Reply:
Maybe it's worth adding a comment stating that.